### PR TITLE
Feature/semantics

### PR DESCRIPTION
**Pull request by hakuturu583**
## Types of PR

- [x] New Features
- [ ] Upgrade of existing features
- [ ] Bugfix

## Description

Enable publish NPC semantic information such as `MOTORCYCLE`, `TRUCK`, `BUS`.
This PR only modified traffic_simulator and simple_sensor_simulator, we need to add features to specify NPC subtype in OpenSCENARIO interpreter.

Currently, subtypes below are supported.
```
UNKNOWN
CAR
TRUCK
BUS
TRAILER
MOTORCYCLE
BICYCLE
PEDESTRIAN
```

## How to review this PR.

Please check passing all CI test cases.

### CODE REVIEW

File: simulation/traffic_simulator/src/entity/entity_base.cpp [hunk 1]
```diff
@@ -133,7 +134,8 @@ auto EntityBase::getVehicleCommand() const -> std::tuple<
   autoware_auto_control_msgs::msg::AckermannControlCommand,
   autoware_auto_vehicle_msgs::msg::GearCommand>
 {
-  THROW_SIMULATION_ERROR("get vehicle command does not support in ", type, " entity type");
+  THROW_SIMULATION_ERROR(
+    "get vehicle command does not support in ", getEntityTypename(), " entity type");
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/entity/entity_base.cpp:138]:
```
"`getVehicleCommand` is not provided for ", getEntityTypename(), " type entity."
```

File: openscenario/openscenario_interpreter/src/syntax/misc_object.cpp [hunk 2]
```diff
@@ -35,8 +35,11 @@ MiscObject::operator traffic_simulator_msgs::msg::MiscObjectParameters() const
 {
   traffic_simulator_msgs::msg::MiscObjectParameters misc_object_parameters;
   {
-    misc_object_parameters.misc_object_category = boost::lexical_cast<String>(misc_object_category);
     misc_object_parameters.name = name;
+    /**
+     * @todo Currently, I use default value for misc object.
+     */
```
Comment (yamacir-kit) [inline openscenario/openscenario_interpreter/src/syntax/misc_object.cpp:41]: Please indicate who the "I" is.
Also, even though this comment is a TODO comment, it does not indicate what needs to be done in the future.

What should be written here in the first place is not "using default values".
It is "WHY you use default values, WHY you think it is a problem, and WHAT you need to do in the future to fix it.

File: openscenario/openscenario_interpreter/src/syntax/pedestrian.cpp [hunk 3]
```diff
@@ -37,7 +37,10 @@ Pedestrian::operator traffic_simulator_msgs::msg::PedestrianParameters() const
   traffic_simulator_msgs::msg::PedestrianParameters parameter;
   {
     parameter.name = name;
-    parameter.pedestrian_category = boost::lexical_cast<String>(pedestrian_category);
+    /**
+     * @todo Currently, I use default value for pedestrian
+     */
```
Comment (yamacir-kit) [inline openscenario/openscenario_interpreter/src/syntax/pedestrian.cpp:42]: Please indicate who the "I" is.
Also, even though this comment is a TODO comment, it does not indicate what needs to be done in the future.

What should be written here in the first place is not "using default values".
It is "WHY you use default values, WHY you think it is a problem, and WHAT you need to do in the future to fix it.

File: openscenario/openscenario_interpreter/src/syntax/vehicle.cpp [hunk 4]
```diff
@@ -37,7 +37,10 @@ Vehicle::operator traffic_simulator_msgs::msg::VehicleParameters() const
   traffic_simulator_msgs::msg::VehicleParameters parameter;
   {
     parameter.name = name;
-    parameter.vehicle_category = boost::lexical_cast<String>(vehicle_category);
+    /**
+     * @todo Currently, I use default value for vehicle.
+     */
```
Comment (yamacir-kit) [inline openscenario/openscenario_interpreter/src/syntax/vehicle.cpp:42]: Please indicate who the "I" is.
Also, even though this comment is a TODO comment, it does not indicate what needs to be done in the future.

What should be written here in the first place is not "using default values".
It is "WHY you use default values, WHY you think it is a problem, and WHAT you need to do in the future to fix it.

**Review by yamacir-kit — CHANGES_REQUESTED** [review]

File: openscenario/openscenario_interpreter/src/syntax/vehicle.cpp (hunk 4 above)
Comment (hakuturu583) [reply openscenario/openscenario_interpreter/src/syntax/vehicle.cpp]: I think this TODO comment is useless.
So, I deleted in this PR.
https://github.com/tier4/scenario_simulator_v2/pull/726/commits/30ee275090b1dfe2079c5b55ff014325844d079b

File: openscenario/openscenario_interpreter/src/syntax/pedestrian.cpp (hunk 3 above)
Comment (hakuturu583) [reply openscenario/openscenario_interpreter/src/syntax/pedestrian.cpp]: I think this TODO comment is useless.
So, I deleted in this PR.
https://github.com/tier4/scenario_simulator_v2/pull/726/commits/30ee275090b1dfe2079c5b55ff014325844d079b

File: openscenario/openscenario_interpreter/src/syntax/misc_object.cpp (hunk 2 above)
Comment (hakuturu583) [reply openscenario/openscenario_interpreter/src/syntax/misc_object.cpp]: I think this TODO comment is useless.
So, I deleted in this PR.
https://github.com/tier4/scenario_simulator_v2/pull/726/commits/30ee275090b1dfe2079c5b55ff014325844d079b

File: simulation/traffic_simulator/src/entity/entity_base.cpp (hunk 1 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/entity/entity_base.cpp]: modified in https://github.com/tier4/scenario_simulator_v2/pull/726/commits/0b70e1db112575734550b61484c7580437177566

**Review by yamacir-kit — APPROVED** [review]

**Pull request merged**